core/DbHelper.php

```diff
@@ -204,6 +204,42 @@ public static function tableHasIndex($table, $indexName)
         return !empty($result);
     }
 
+    /**
+     * Returns the default database charset to use
+     *
+     * Returns utf8mb4 if supported, with fallback to utf8
+     *
+     * @return string
+     * @throws Tracker\Db\DbException
+     */
+    public static function getDefaultCharset()
+    {
+        $result = Db::get()->fetchRow("SHOW CHARACTER SET LIKE 'utf8mb4'");
+
+        if (empty($result)) {
+            return 'utf8'; // charset not available
+        }
+
+        $result = Db::get()->fetchRow("SHOW VARIABLES LIKE 'character_set_database'");
+
+        if (!empty($result) && $result['Value'] === 'utf8mb4') {
+            return 'utf8mb4'; // database has utf8mb4 charset, so assume it can be used
+        }
+
+        $result = Db::get()->fetchRow("SHOW VARIABLES LIKE 'innodb_file_per_table'");
+
+        if (empty($result) || $result['Value'] !== 'ON') {
+            return 'utf8'; // innodb_file_per_table is required for utf8mb4
+        }
```

Comment:
Should we cache the result of this function? And haven't been following this issue, but do we need to do the `innodb_file_per_table` check if `character_set_database` is set to `utf8mb4`, or can we just assume it will work?

Reply:
Or maybe we can set `Config::getInstance()->database['charset']` if it's not set to the default...

Reply:
Actually I see this is already done during installation/update, so maybe that's enough

Reply:
Was wondering the same if we need to check for `SHOW VARIABLES LIKE 'character_set_database'` after `innodb_file_per_table `?